diff --git a/cc/cicada/transaction.cc b/cc/cicada/transaction.cc
--- a/cc/cicada/transaction.cc
+++ b/cc/cicada/transaction.cc
@@ -646,8 +646,8 @@
   if (cap_floor <= __atomic_load_n(&ThreadRtsArray[thid].obj_,
                                     __ATOMIC_ACQUIRE))
     GC_INC(thid, cap_no_raise);
 #endif
-  gc_publish_rts(thid, cap_floor);
+  gc_publish_rts(thid, target - 1);
 #line 411
   return true;
 }
